plugin/symbols.py

```diff
@@ -119,28 +126,27 @@ def process_symbols(self, items: List[Dict[str, Any]]) -> List[List[str]]:
 
     def process_document_symbols(self, items: List[Dict[str, Any]]) -> List[List[str]]:
         quick_panel_items = []  # type: List[List[str]]
-        self.found_at_least_one_nonempty_detail = False
+        names = []  # type: List[str]
         for item in items:
-            self.process_document_symbol_recursive(quick_panel_items, item, 0)
-        if self.found_at_least_one_nonempty_detail:
-            return quick_panel_items
-        else:
-            return [[item[0], item[1]] for item in quick_panel_items]
+            self.process_document_symbol_recursive(quick_panel_items, item, names)
+        return quick_panel_items
 
     def process_document_symbol_recursive(self, quick_panel_items: List[List[str]], item: Dict[str, Any],
-                                          depth: int) -> None:
+                                          names: List[str]) -> None:
         kind = item['kind']
         self.regions.append((range_to_region(Range.from_lsp(item['range']), self.view),
                              range_to_region(Range.from_lsp(item['selectionRange']), self.view),
                              format_symbol_scope(kind)))
-        name = ' ' * (4 * depth) + item['name']
-        quick_panel_item = [name, format_symbol_kind(kind), item.get('detail') or '']
-        if quick_panel_item[2]:
-            self.found_at_least_one_nonempty_detail = True
-        quick_panel_items.append(quick_panel_item)
-        children = item.get('children') or []
-        for child in children:
-            self.process_document_symbol_recursive(quick_panel_items, child, depth + 1)
+        name = item['name']
+        with _additional_name(names, name):
+            kind_and_detail = format_symbol_kind(kind)
+            detail = item.get('detail')
+            if detail:
+                kind_and_detail += " - {}".format(detail)
+            quick_panel_items.append([name, " → ".join(names), kind_and_detail])
```

Comment:
There is some redundancy for top-level symbols but no big deal.

![Screenshot 2020-07-18 at 15 46 26](https://user-images.githubusercontent.com/153197/87853834-eca1b580-c90d-11ea-834f-1b5d4716f0bb.png)